contributors/design-proposals/api-machinery/admission-webhook-GA-proposal.md

```diff
@@ -0,0 +1,488 @@
+# Graduate Admission Webhooks to GA
+
+Authors: @mbohlool
+
+Reviewers: @liggit @deads2k @sttts @caesarxuchao
+
+Sig: api-machinery
+
+## Summary
+
+Admission webhooks are a way to extend kubernetes by putting hooks over object creation/modification. Admission webhooks can mutate or validate the object. This feature has been Beta since Kubernetes 1.9. This document outline required steps to graduate it to GA.
+
+## New Features
+
+Based on users feedback, these are the changes that are proposed to graduate Admission Webhooks to GA.
+
+### Object selector
+
+Currently Admission Webhook supports namespace selectors, but that may not be enough for some cases that admission webhook need to be skipped on some objects. For example if the Admission Webhook is running inside cluster and its rules includes wildcards which match required API objects for its own execution. An object selector would be useful exclude those objects. Also in case of an optional webhooks, an object selector gives the end user to include or exclude an object without having access to admission webhook configuration which is normally restricted to cluster admins.
+Note that namespace objects must match both object selector (if specified) and namespace selector to be sent to the Webhook except for cluster-scoped resources which namespace selector will not applied to. The propsoe change is to add an ObjectSelector to the webhook API both in v1 and v1beta1.
+
+```golang
+type Webhook struct {
+    ...
+     // ObjectSelector decides whether to run the webhook on an object based
+     // on whether the object.metadata.labels matches the selector. An object is
```

Comment:
> An object is matches both NamespaceSelector and ObjectSelector.

I think this was supposed to read: "A namespace object must match both namespaceSelector (if present) and objectSelector (if present)."